src/security/KeyData.cc

```diff
@@ -50,11 +50,13 @@ Security::KeyData::loadCertificates()
         while (const auto bundledCert = Ssl::ReadOptionalCertificate(bio)) {
             assert(!wireCerts.empty()); // this->cert is there (at least)
 
-            // We cannot chain any certificate after a self-signed certificate. This
-            // check also protects the IssuedBy() check below from adding duplicated
-            // (i.e. listed multiple times in the bundle) self-signed certificates.
-            if (SelfSigned(*wireCerts.back())) {
-                debugs(83, DBG_PARSE_NOTE(2), "WARNING: Ignoring certificate after a self-signed one: " << *bundledCert);
+            // We ignore a self-signed certificate because it should not be
+            // sent: The recipients that do not already have it should not trust
```

Comment:
This is a more significant behavioural change than it seems. When I attempted this a few years back there was immediate feedback about some environments failing to validate the TLS connections to Squid.

I am not against another attempt to fix this, but please do update the squid.conf documentation for all directives using tls-cert= options and also Squid-6 release notes for their behaviour change.